scripts/build

```diff
@@ -5,5 +5,27 @@ source $(dirname $0)/version
 
 cd $(dirname $0)/..
 
-mkdir -p bin
-CGO_ENABLED=0 go build -ldflags "-X main.VERSION=$VERSION -linkmode external -extldflags -static -s" -o bin/agent
+go_build() {
+  echo Building for os=$1 arch=$2
+  mkdir -p bin/$1/$2
+  if [ "$1" == "windows" ]; then
+    o=bin/$1/$2/agent.exe
+  else 
+    o=bin/$1/$2/agent
+  fi
+  if [ "$1" == "linux" ]; then
+    ldflags="-X main.VERSION=$VERSION -linkmode external -extldflags -static -s"
+  elif [ "$1" == "windows" ]; then
+    ldflags="-X main.VERSION=$VERSION -extldflags -static -s"
+  else
+    ldflags="-X main.VERSION=$VERSION"
+  fi
+  GOOS=$1 GOARCH=$2 CGO_ENABLED=0 go build -ldflags "$ldflags" -o $o
+}
+
+if [ "$CROSS" == "true" ]; then
+  go_build linux amd64
+  go_build windows amd64
+else
+  go_build linux amd64
+fi
```

Comment:
Nit: We have a cross-compile script pattern in https://github.com/rancher/cli/blob/v1.6/scripts/build so i think it might be better to reuse this pattern.